File: src/main/java/fr/centralesupelec/sio/data/DummyAccountsRepository.java
package fr.centralesupelec.sio.data;

import fr.centralesupelec.sio.model.Account;
import fr.centralesupelec.sio.model.Movie;

import java.util.Arrays;

public class DummyAccountsRepository extends AccountsRepository {

    @Override
    public Account getAccount(String username) {
        Account a1 = new Account();
        a1.setUsername("admin@ecp.sio.fr");
        a1.setPasswordHash("toto");
        return a1.getUsername().equals(username) ? a1 : null;
    }

}

File: src/main/java/fr/centralesupelec/sio/data/DummyAccountsRepository.kt
package fr.centralesupelec.sio.data

import fr.centralesupelec.sio.model.Account

/**
 * A concrete [AccountsRepository] backed by an in-memory list of static [Account] entities.
 */
internal class DummyAccountsRepository : AccountsRepository {

    private val accounts: List<Account> = listOf(
            Account(username = "admin@ecp.sio.fr", passwordHash = "password")
    )

    override fun getAccount(username: String): Account? =
        accounts.firstOrNull { it.username.equals(username, ignoreCase = true) }

}
